utils/python/CIME/user_mod_support.py

```diff
@@ -56,6 +56,8 @@ def apply_user_mods(caseroot, user_mods_path, ninst=None):
                             expect(False, "Could not write file %s in caseroot %s"
                                    %(case_source_mods,caseroot))
 
+    include_dirs.reverse()
```

Comment:
Did you test this?   I think that without an assignment in line 59 it doesn't do anything.   You could also write line 60 as :
for include_dir in include_dirs.reverse()

I prefer this change to the one in 757